src/cargo/util/lints.rs

```diff
@@ -157,56 +170,61 @@ pub fn check_implicit_features(
     error_count: &mut usize,
     gctx: &GlobalContext,
 ) -> CargoResult<()> {
-    let lint_level = IMPLICIT_FEATURES.level(lints, pkg.manifest().edition());
+    let edition = pkg.manifest().edition();
+    if edition >= Edition::Edition2024 {
+        return Ok(());
+    }
```

Comment:
This should likewise probably have a comment like
```suggestion
    let edition = pkg.manifest().edition();
    // In Edition 2024+, instead of creating optional features, the dependencies are unused.  See [`UNUSED_OPTIONAL_DEPENDENCY`]
    if edition >= Edition::Edition2024 {
        return Ok(());
    }
```